wormhole_chain/x/wormhole/keeper/msg_server_wasmd_test.go

```diff
@@ -34,34 +31,29 @@ func createWasmInstantiatePayload(code_id uint64, label string, json_msg string)
 	// - code_id (big endian)
 	// - label
 	// - json_msg
-	hash_base := make([]byte, 8)
-	binary.BigEndian.PutUint64(hash_base, code_id)
-	hash_base = append(hash_base, []byte(label)...)
-	hash_base = append(hash_base, []byte(json_msg)...)
 	var expected_hash [32]byte
 	keccak := sha3.NewLegacyKeccak256()
-	keccak.Write(hash_base)
+	binary.Write(keccak, binary.BigEndian, code_id)
+	keccak.Write([]byte(label))
+	keccak.Write([]byte(json_msg))
 	keccak.Sum(expected_hash[:0])
 
-	payload := []byte{}
-	payload = append(payload, keeper.WasmdModule[:]...)
-	payload = append(payload, byte(keeper.ActionInstantiateContract))
-	chain_bz := [2]byte{}
-	binary.BigEndian.PutUint16(chain_bz[:], uint16(WH_CHAIN_ID))
-	payload = append(payload, chain_bz[:]...)
+	payload := bytes.NewBuffer(keeper.WasmdModule[:])
```

Comment:
The [documentation](https://pkg.go.dev/bytes#NewBuffer) for `bytes.NewBuffer` says:

> The new Buffer takes ownership of buf, and the caller should not use buf after this call.

So I think we need to make a copy here.  Or do something like:

```suggestion
	var payload bytes.Buffer
	payload.Write(keeper.WasmdModule[:])
```

Reply:
Hmm, I wonder if committing the suggestion will dismiss the PR approval.

Reply:
it did :)